### DISCUSSION TITLE
SEO and Vuex with pre-rendering

### DISCUSSION THREAD

**Discussion by giticon** [nuxt/nuxt#8214] [General]
### Scenario

I have an indexblog.vue page with this code
```
<div id="blogListOut">
	<blog-out />
</div>
```

The "blog-out" component contains a loop that prints a list of articles for a blog

```
<div v-for="article of articles" :key="article.slug">
	<nuxt-link :to="{ name: 'blog-slug', params: { slug: article.slug } }">
	<div class="article_list">
	<div class="article_list_title"><h2>{{ article.title }}}</h2></div>
	...
```

through a function

```
computed:{
    articles() {
      return this.$store.state.searchArticles.articles
    },
```

Vuex fills the array correctly

```
export const state = () => ({
	articles: [],
})

async startArticles ({ commit }) {
	    const articles = await this.$content('articles')
	    .sortBy('createdAt','desc')
        .fetch()
	    await commit('searchEffective', articles)
	},
...

export const mutations = {
	searchEffective(state, payload) {
		state.articles = payload 
	},
```

this is ok.

The point is that the result in html is inserted in the virtual DOM and the resulting html is inserted virtually in 

`<div id="blogListOut"><div>`

The page source does not contain the html of the post list (CTRL+U). It is visible only with the inspector (F12).

The question: is it possible to make Nuxt/Vuex return the list of posts in a static way (raw html) instead of dynamically hooking the result to the div?

That is: is it possible to pre-rendering data coming from Vuex if they are called by a component?

**Comment by TheAlexLichter** [comment]
If you fill the Vuex store, e.g. via `middleware` or `nuxtServerInit` then the data will be ready on the server side and included in the rendered HTML.

**Reply by giticon** [reply]
tnx @manniL 

in the nuxtjs doc 

"If the action nuxtServerInit is defined in the store **and the mode is universal**, Nuxt.js will call it with the context (only from the server-side)."

In nuxt.config.js I have

`export default {
  target: 'static',`

(vercel, static deployment)

is still valid in your opinion?

**Comment by giticon** [comment]
I write this in case someone needs it

```js
// nuxt.config.js
export default {
  target: 'server',
  ....
```

```js
// menunav.vue
watch: {
    '$route' () {
        if(this.open === true){
          this.open = false
        }
        this.$store.dispatch('searchArticles/startArticles')
    }
  },
```

```js
// store/index.js
export const actions = {
  async nuxtServerInit ({ dispatch }) {
    await dispatch('searchArticles/startArticles')
  }
}
```

This avoids getting a blank page the first time load it.

I also confirm that switching from "static" mode to "server" everything works (Vercel hosting) without extra configuration.
